src/main.ts

```diff
@@ -6,9 +6,32 @@ import {
 } from '@us-epa-camd/easey-common/nestjs';
 import { useContainer } from 'class-validator';
 import { ValidationPipe } from '@nestjs/common';
+import * as http from 'http';
 
 import { AppModule } from './app.module';
 
+let server: http.Server;
+
+async function gracefulShutdown(exitCode: number) {
+  console.error('Initiating graceful shutdown...');
+
+  if (server) {
+    await new Promise<void>((resolve) => {
+      server.close(() => {
+        console.log('HTTP server closed');
+        resolve();
+      });
+    });
+  }
+
+  setTimeout(() => {
+    console.error('Forced shutdown after timeout');
+    process.exit(exitCode);
+  }, 10000).unref();
+
+  process.exit(exitCode);
+}
```

Comment:
Since `process.exit()` is called immediately after scheduling the timeout, the timeout is redundant. I think, if we want the timeout to force a shutdown if `server.close()` hangs, then something like the following should be used:
```suggestion
async function gracefulShutdown(exitCode: number) {
  console.error('Initiating graceful shutdown...');

  // Force shutdown if graceful shutdown hangs
  const forceExitTimer = setTimeout(() => {
    console.error('Forced shutdown after timeout');
    process.exit(exitCode);
  }, 10000);
  forceExitTimer.unref();

  if (server) {
    await new Promise<void>((resolve) => {
      server.close(() => {
        console.log('HTTP server closed');
        resolve();
      });
    });
  }

  // If we get here, graceful shutdown completed
  clearTimeout(forceExitTimer);
  process.exit(exitCode);
}
```